### ISSUE TITLE
Viewing 3D-objects?

### ISSUE THREAD

**Issue by QuietOcean** [owncloud/files_mediaviewer#233]
Dear Members,

Kindly I would like to ask if I could propose the idea of visualizing 3D-objects in files such as .stl.
With 3D-printing and design software increasing in use it would make it easier to scroll through files and selecting those you need for a particular project. Arguable one could install the desktop client to sync the files but it is not always handy or fast. I was wondering what kind of work would be required to realise this idea... is just Owncloud app that requires adjustments or does the browser (like Firefox) also need an update?

**Comment by j-ed** [comment]
If you have a look on the [Nextcloud Apps store](https://apps.nextcloud.com/), you will see that a [3D viewer app](https://apps.nextcloud.com/apps/files_3d) already exists, which might be what you're looking for. Unfortunately it currently only supports Nextcloud up to v16.
If you have a look on the linked [Github repository](https://github.com/v1r0x/files_3d/) of the app, you will see that the developer is still applying changes to the code. Therefore I would assume that he's working on a new release of his app.

**Comment by QuietOcean** [comment]
Thank you for your kind reply. I forwarded my question to the app developer of [3D viewer](https://github.com/v1r0x/files_3d/issues/11). I have to admit that it does trigger the question whether Nextcloud or Owncloud is more suitable for us.

**Comment by v1r0x** [comment]
Hey there,

I have several nc instances running, that's why I develop my apps for nc.
Since the split of oc and nc I don't have a oc instance anymore and have since then never worked with one. In the polls app we had builds for both systems, but none of the devs was using oc. Thus, testing and adding features was quite difficult. We decided to drop oc support, as long no active oc user helps with modifications needed for oc.
For the 3d app I would suggest the same. I sometimes have problems with the api of nc. Fiddling with two apis leads to more maintenance and less fun. I also had huge problems to publish apps to the oc appstore and didn't want to accept their ridiculous restrictions.

Long story short:
I won't support oc by myself for several reasons. But if someone is willing to port the code (if needed) in their own repo and publish it for oc, I'm totally fine with that and will try to help them with any feature-related questions.

Stay safe!

**Comment by j-ed** [comment]
At first I would recommend to never bind a decision if Nextcloud or ownCloud is more suitable for your needs to the availability of a 3rd-party 3D viewer. If you want to use a private cloud for your business I would always compare all bits and pieces and also contact Nextcloud GmbH to get an idea of the professional user base and the available support model etc.. Perhaps they would cover that app in their proposal too, so that you're on the save side.

**Comment by QuietOcean** [comment]
Thank you j-ed for your reply. I agree. We found a work around by using WebDAV to visualise objects locally on a computer. Not ideal but it works. Thank you v1r0x for the explanation. Good luck & stay safe.

**Comment by alexstocker** [comment]
You may check this out https://marketplace.owncloud.com/apps/files_3dviewer
